### PR TITLE
Protect: Use --spacing-base property for layout and positioning styles

### PR DESCRIPTION
**Pull request by nateweller**
#### Changes proposed in this Pull Request:

* This PR removes the fixed `px` values used for `margin`, `padding`, and `position` styles in Protect, in favor of the `--spacing-base` CSS property defined by the `ThemeProvider` component.

* Additionally, this PR enforces consistency in how CSS variables are formatted (i.e. `var( --spacing-base )` instead of `var(--spacing-base)`.

* Each style that uses `--spacing-base` includes a comment describing the default pixel value(s) calculated in that line.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

1201069996155224-as-1203163359449358

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Build the branch and validate there is no regression in spacing of elements.

### CODE REVIEW

File: projects/plugins/protect/src/js/components/summary/styles.module.scss
Comment (dkmyta) [inline projects/plugins/protect/src/js/components/summary/styles.module.scss]: Looks like we've got an extra `margin-left` declaration here that can be removed. I _think_ that should fix the current `plugins/protect` build issues as well.

**Review by dkmyta — CHANGES_REQUESTED** [review]
Looks great - one minor change is needed that should address the failing PR check.

File: projects/plugins/protect/src/js/components/modal/styles.module.scss
Comment (dkmyta) [inline projects/plugins/protect/src/js/components/modal/styles.module.scss]: This calculation doesn't appear to be working so the `padding-top` isn't being applied:

Screen Shot 2022-11-02 at 12 30 42

File: projects/plugins/protect/src/js/components/admin-page/styles.module.scss
Comment (dkmyta) [inline projects/plugins/protect/src/js/components/admin-page/styles.module.scss]: I wasn't able to figure out an easy way to get this column to display, but my assumption is that applying an additional `px` value this way won't return the expected results as seen with the styling of the Modal component.

File: projects/plugins/protect/src/js/components/summary/styles.module.scss
Comment (nateweller) [reply projects/plugins/protect/src/js/components/summary/styles.module.scss]: Thanks for catching this, resolved via [273abcb](https://github.com/Automattic/jetpack/pull/27175/commits/273abcb2ec64612e3370a79c347047c90ac4221e)!

File: projects/plugins/protect/src/js/components/modal/styles.module.scss
Comment (nateweller) [reply projects/plugins/protect/src/js/components/modal/styles.module.scss]: Turns out I needed to specify the `px` unit for the `+ 2` here! Resolved via [5f31b9b](https://github.com/Automattic/jetpack/pull/27175/commits/5f31b9b1c3cbd454730415f7adc4004a42e87adf) :+1:

File: projects/plugins/protect/src/js/components/admin-page/styles.module.scss
Comment (nateweller) [reply projects/plugins/protect/src/js/components/admin-page/styles.module.scss]: Same issue as the Modal padding, added a `px` unit via [5f31b9b](https://github.com/Automattic/jetpack/pull/27175/commits/5f31b9b1c3cbd454730415f7adc4004a42e87adf).

**Comment by nateweller** [comment]
Thanks for the review @dkmyta! Issues have been addressed.

**Review by dkmyta — APPROVED** [review]

**Comment by dkmyta** [comment]
Nice! Good to go now once the merge conflicts are resolved 👍🏻

**Pull request merged**
